r2-netty/src/main/java/com/linkedin/r2/netty/client/http2/Http2ChannelLifecycle.java

```diff
@@ -221,7 +221,7 @@ public void onSuccess(Channel channel)
           }
         });
       }
-    }, new ObjectCreationTimeoutException(
+    }, () -> new ObjectCreationTimeoutException(
     "Exceeded creation timeout of " + _channelCreationTimeoutMs + "ms: for HTTP/2 parent channel, remote=" + _address)));
```

Comment:
Also, note that this code path only gets invoked when there are no spare channels in the pool for the incoming request. This is not invoked for every request!

Reply:
Agree the save here is small. It's just one of the spots that our customer identified.